### PR TITLE
Bugfix/add new space

### PR DESCRIPTION
**Pull request by msadura**
Fix for blank page when creating workspace from sidebar.

Additional stuff:
- `useSpaces` will now return object instead of an array
- `createNewSpace` - moved to `useSpaces` to have it in a single place to reuse

### CODE REVIEW

File: hooks/useSpaces.tsx [hunk 1]
```diff
@@ -33,7 +40,19 @@ export function SpacesProvider ({ children }: { children: ReactNode }) {
...
 
-  const value = useMemo(() => [spaces, setSpaces, isLoaded] as IContext, [spaces, isLoaded]);
+  const createNewSpace = useCallback(async (newSpace: Prisma.SpaceCreateInput) => {
+    const space = await charmClient.createSpace(newSpace);
+    setSpaces((s) => [...s, space]);
+    // refresh user permissions
+    const _user = await charmClient.getUser();
+    setUser(_user);
+    // give some time for spaces state to update or user will be redirected to /join in RouteGuard
+    setTimeout(() => {
```
Comment (msadura) [inline hooks/useSpaces.tsx:50]: lack of this timeout in one of the components caused the issue 😓

File: package.json [hunk 2]
```diff
@@ -16,7 +16,7 @@
     "migrate:prod": "prisma migrate deploy",
     "migrate:staging": "npx dotenv -e .env.staging -- prisma migrate deploy",
     "server:dev": "next dev",
-    "debug:e2e": "dotenv -e .env.test.local -- npx playwright test --headed --timeout=0",
+    "debug:e2e": "dotenv -e .env.test.local -- npx playwright test **YOUR_TEST_PATH.spec.ts** --headed --timeout=0",
```
Comment (mattcasey) [inline package.json:19]: i think this makes the command useless? YOu can just add the file at the end. What i've been doing is:
```
npm run debug:e2e __e2e__/login.spec.ts
```
Should we add this to the README?

File: hooks/useSpaces.tsx [hunk 3]
```diff
@@ -1,19 +1,26 @@
...
 import type { ReactNode } from 'react';
-import { createContext, useContext, useEffect, useMemo, useState } from 'react';
+import { useCallback, createContext, useContext, useEffect, useMemo, useState } from 'react';
 
 import charmClient from 'charmClient';
 
 import { useUser } from './useUser';
 
-type IContext = [spaces: Space[], setSpaces: (spaces: Space[]) => void, isLoaded: boolean];
+type IContext = {
```
Comment (mattcasey) [inline hooks/useSpaces.tsx:10]: thanks for the cleanup of array => map

File: __e2e__/sidebarAddNewSpace.spec.ts [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+    // 1. Make sure there is a button to add a new workspace
+    const addNewSpaceBtn = page.locator('data-test=sidebar-add-new-space');
+    await expect(addNewSpaceBtn).toBeVisible();
+
+    // 2. Open create space dialod
+    await addNewSpaceBtn.click();
+
+    await expect(page.locator('data-test=create-space-form')).toBeVisible();
+
+    const nameInput = page.locator('data-test=workspace-name-input >> input');
```
Comment (mattcasey) [inline __e2e__/sidebarAddNewSpace.spec.ts:59]: For e2e, I want to be EXTREMELY minimal on what we are testing. Anything that can be done via Cypress or Jest should be done there instead, for example testing that the domain is updated when the name changes should be done in Cypress. I think we should remove all the logic around inputs in this test. Just open the modal and click 'create workspace' and make sure that we get to an actual page on the other side

**Review by mattcasey — CHANGES_REQUESTED** [review]
one request: reduce the scope of the e2e test

File: package.json (hunk 2 above)
Comment (msadura) [reply package.json:19]: ah I though you need manually put you filename in there 😅 I will fix that

**Comment by msadura** [comment]
@mattcasey  updates pushed - removed input tests, added readme info

**Review by mattcasey — APPROVED** [review]

**Pull request merged**